scripts/cluster.sh

```diff
@@ -215,6 +215,220 @@ function download_iso() {
+    #   $1 - cluster_name: Name of the cluster (defaults to CLUSTER_NAME from env)
+    #   $2 - node_type: "master" or "worker" (default: master)
+    #   $3 - action: "url" or "download" (default: download)
+    #   $4 - download_path: Path to download ISO (optional)
+    #   $5 - iso_type: "minimal" or "full" (default: minimal)
+    
+    local cluster_name="${1:-${CLUSTER_NAME}}"
+    local node_type="${2:-master}"
+    local action="${3:-download}"
+    local download_path="${4:-${ISO_FOLDER}}"
+    local iso_type="${5:-${ISO_TYPE:-minimal}}"  # Get from param or ENV, default to minimal
+    local day2_cluster_name="${cluster_name}-day2"
+    
+    # Normalize ISO type by removing any -iso suffix
+    iso_type="${iso_type%-iso}"
+    
+    # Basic validation
+    if [ -z "${cluster_name}" ]; then
+        log "ERROR" "Cluster name is required"
+        exit 1
+    fi
+    
+    # For worker nodes, we need a day2 cluster
+    if [ "${node_type}" = "worker" ]; then
+        # Make sure the day2 cluster exists
+        if ! aicli info cluster ${day2_cluster_name} >/dev/null 2>&1; then
+            log "INFO" "Creating day2 cluster for worker nodes"
+            
+            # Get original cluster version for consistency
+            local original_version=$(aicli info cluster ${cluster_name} -f openshift_version -v 2>/dev/null)
+            if [ -z "${original_version}" ]; then
+                log "ERROR" "Could not determine OpenShift version of the original cluster"
+                exit 1
+            fi
+            
+            # Create day2 cluster - don't specify ISO type to use default
+            aicli create cluster \
+                -P openshift_version="${original_version}" \
+                -P base_dns_domain="${BASE_DOMAIN}" \
+                -P pull_secret="${OPENSHIFT_PULL_SECRET}" \
+                -P public_key="${SSH_KEY}" \
+                -P day2=true \
+                "${day2_cluster_name}" >/dev/null
+                
+            log "INFO" "Day2 cluster ${day2_cluster_name} created successfully"
+        fi
+        
+        # Handle worker node ISO
+        if [ "${action}" = "url" ]; then
+            # Get ISO URL for workers
+            log "INFO" "Getting ISO URL for worker nodes (ISO type: ${iso_type})"
+            
+            # Get ISO URL using infraenv method (more reliable)
+            local infraenv_id=$(aicli list infraenvs 2>/dev/null | grep "${day2_cluster_name}" | awk -F'|' '{print $2}' | tr -d ' \t' | head -1)
+            local iso_url=""
+            
+            if [ -n "${infraenv_id}" ]; then
+                iso_url=$(aicli info infraenv "${infraenv_id}" 2>/dev/null | grep -E "(download_url|iso_download_url):" | awk '{print $2}')
+            fi
+            
+            # Fallback to direct method if infraenv failed
+            if [ -z "${iso_url}" ]; then
+                iso_url=$(aicli info iso "${day2_cluster_name}" 2>/dev/null | grep -i "url:" | awk '{print $2}')
```

Comment:
This is totally enough to download iso, please remove all other options. No need to fallback